climakitae/threshold_tools.py

```diff
@@ -884,3 +889,523 @@ def return_period(ams):
+        self.selections.variable = "Air Temperature at 2m"
+
+        self.units2 = self.selections.descrip_dict[self.selections.variable]["native_unit"]
+
+        # Location defaults
+        # self.location.area_subset = 'states'
+        # self.location.cached_area = 'CA'
+        self.location.area_subset = 'CA counties'
+        self.location.cached_area = 'Sacramento County'
+
+        # Get the underlying dataarray
+        self.da = _read_from_catalog(selections = self.selections, location = self.location, cat = self._cat).compute()
+
+        self.threshold_value = round(self.da.mean().values.item())
+        self.param.threshold_value.label = f"Value (units: {self.da.units})"
+
+        self.param["variable2"].objects = self.selections.param.variable.objects
+        self.param["cached_area2"].objects = self.location.param.cached_area.objects
+
+    reload_plot = param.Action(lambda x: x.param.trigger('reload_plot'), label='Reload Plot')
+
+    # For reloading data
+    reload_data = param.Action(lambda x: x.param.trigger('reload_data'), label='Reload Data')
+    changed_loc_and_var = param.Boolean(default=True)
+    changed_units = param.Boolean(default=False)
+
+    @param.depends("area_subset2","cached_area2","variable2", watch=True)
+    def _updated_bool_loc_and_var(self):
+        """Update boolean if any changes were made to the location or variable"""
+        self.changed_loc_and_var = True
+
+    @param.depends("units2", watch=True)
+    def _updated_units(self):
+        """Update boolean if a change was made to the units"""
+        self.changed_units = True
+
+    @param.depends("variable2","units2",watch=True)
+    def _update_unit_options(self): 
+        """ Update unit options and native units for selected variable. """
+        _default_unit = self.selections.descrip_dict[self.variable2]["native_unit"]
+        _alt_units = self.selections.descrip_dict[self.variable2]["alt_unit_options"]
+        if pd.isna(_alt_units): 
+            self.param["units2"].objects = [_default_unit]
+        else:
+            self.param["units2"].objects = _default_unit.split(", ")+_alt_units.split(", ")
+        if self.units2 not in self.param["units2"].objects:
+            self.units2 = _default_unit
+
+    @param.depends("reload_data", watch=True)
+    def _update_data(self):
+        """If the button was clicked and the location, variable, or units were 
+        changed, reload the data from AWS"""
+        if self.changed_loc_and_var:
+            self.da = _read_from_catalog(selections = self.selections, location = self.location, cat = self._cat).compute()
+            self.changed_loc_and_var = False
+        if self.changed_units:
+            self.da = _convert_units(da=self.da, selected_units=self.units2)
+            self.threshold_value = round(self.da.mean().values.item())
+            self.param.threshold_value.label = f"Value (units: {self.da.units})"
+            self.changed_units = False
```

Comment:
It's not clear to me what the purpose of the unit options are. If they are just used in computation, it shouldn't matter if you convert units, because the output plot doesn't show the units from the original variable anywhere (just count vs. time). And, even when I change units (say, from K to degF), the thresholds panel still shows the data in K. 

Unless I'm missing something (very possible), I would suggest removing the unit conversion option all together. 

Or, is it there for the purpose of data downloading, after the panel? If so, I still think it's clearer to just remove the units option all together, unless you can loop it in with the panel options entirely. 

<img width="700" alt="units mismatch" src="https://user-images.githubusercontent.com/66140951/192852320-b4ceef7d-5e35-497c-98d3-05ecf860073d.png">

Reply:
Yep, that works!

Reply:
It might make more sense to include `_exceedance_visualize` as part of `ThresholdDataParams`. That way, you could have the value shown (i.e. Value (units: K)) update in a function with `param.depends("units",watch=True)` so it dynamically updates.